_search-plugins/search-pipelines/ml-inference-search-request.md

```diff
@@ -0,0 +1,532 @@
+
+```json
+POST /_plugins/_ml/models/cleMb4kBJ1eYAeTMFFg4/_predict
+{
+  "parameters": {
+    "inputs": [
+      {
+        ...
+      }
+    ]
+  }
+}
+```
+
+Thus, to use an externally hosted sentiment analysis model, send a Predict API request in the following format:
+
+```json
+POST /_plugins/_ml/models/cywgD5EB6KAJXDLxyDp1/_predict
+{
+  "parameters": {
+    "inputs": "happy moments"
+  }
+}
+```
+{% include copy-curl.html %}
+
+The model processes the input and generates a prediction based on the sentiment of the input text. In this case, the sentiment is positive: 
+
+```json
+{
+  "inference_results": [
+    {
+      "output": [
+        {
+          "name": "response",
+          "dataAsMap": {
+            "label": "POSITIVE",
+            "score": "0.948"
+          }
+        }
+      ],
+      "status_code": 200
+    }
+  ]
+}
+```
+
+When specifying the `input_map` for an externally hosted model, you can directly reference the `inputs` field instead of providing its dot path `parameters.inputs`:
+
+```json
+"input_map": [  
+  {
+    "inputs": "query.term.label.value"
+  }
+]
+```
+
+**Step 2: Run the pipeline**
+
+Once you have created a search pipeline, you can run the same term query with the search pipeline:
+
+```json
+GET /my_index/_search?search_pipeline=my_pipeline_request_review
+{
+  "query": {
+    "term": {
+      "label": {
+        "value": "happy moments",
+        "boost": 1
+      }
+    }
+  }
+}
+```
+{% include copy-curl.html %}
+
+The query term value is rewritten based on the model's output. The model determines that the sentiment of the query term is positive, so the rewritten query looks like the following:
+
+```json
+{
+  "query": {
+    "term": {
+      "label": {
+        "value": "POSITIVE",
+        "boost": 1
+      }
+    }
+  }
+}
+```
+
+The response includes the document whose `label` field has the value `POSITIVE`:
+
+```json
+{
+  "took": 288,
+  "timed_out": false,
+  "_shards": {
+    "total": 1,
+    "successful": 1,
+    "skipped": 0,
+    "failed": 0
+  },
+  "hits": {
+    "total": {
+      "value": 1,
+      "relation": "eq"
+    },
+    "max_score": 0.00009405752,
+    "hits": [
+      {
+        "_index": "my_index",
+        "_id": "3",
+        "_score": 0.00009405752,
+        "_source": {
+          "passage_text": "I am excited",
+          "passage_language": "en",
+          "label": "POSITIVE"
+        }
+      }
+    ]
+  }
+}
+```
+
+### Example: Local model
+
+The following example configures an `ml_inference` processor with a local model to rewrite a term query into a k-NN query.
+
+**Step 1: Create a pipeline**
+
+The following example creates a search pipeline for the `huggingface/sentence-transformers/all-distilroberta-v1` local model. The model is a sentence transformer [pretrained model]({{site.url}}{{site.baseurl}}/ml-commons-plugin/pretrained-models/#sentence-transformers)
+hosted in your OpenSearch cluster.
```

Comment:
Line above: "The model is a pretrained sentence transformer model" (this would be the expected syntax)?